Sources/NIOHTTP2/HTTP2FrameParser.swift

```diff
@@ -405,24 +405,11 @@ struct HTTP2FrameDecoder {
             
         case .simulatingDataFrames(var state):
             // NB: already checked for root stream before entering this state
-            guard state.payload.readableBytes > 0 else {
+            if state.payload.readableBytes == 0 && state.remainingByteCount > 0 {
                 // need more bytes!
                 return .needMoreData
             }
-            
-            if state.remainingByteCount <= Int(state.expectedPadding) {
-                assert(state.flowControlledLength() == 0)
-                // we're just eating pad bytes now, maintaining state and emitting nothing
-                if state.payload.readableBytes >= state.remainingByteCount {
-                    // we've got them all, move to idle state with any following bytes
-                    state.payload.moveReaderIndex(forwardBy: state.remainingByteCount)
-                    self.state = .accumulatingFrameHeader(AccumulatingFrameHeaderParserState(unusedBytes: state.payload))
-                } else {
-                    // stay in state and wait for more bytes
-                    return .needMoreData
-                }
-            }
```

Comment:
Why is this block gone?

Reply:
It's covered by the if-else block below.

Reply:
Ah, actually, I think that's not quite the case when the we don't have enough padded bytes as we'd return a frame with empty bytes instead of `.needMoreData`.

Reply:
Agreed, I think this is a new conditional not covered here at all. We probably need a new if block just after the guard to handle this.